dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java

```diff
@@ -60,86 +64,127 @@ public VTLResource() {
-        final Language language = WebAPILocator.getLanguageWebAPI().getLanguage(request);
-        IPersona persona = null;
-        final Optional<Visitor> visitor = APILocator.getVisitorAPI().getVisitor(request, false);
+    /**
+     * @param request HttpServletRequest
+     * @return Response
+     */
+    @POST
+    @Path("/{folder}/{path: .*}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, MediaType.TEXT_PLAIN})
+    public final Response post(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                                     @Context UriInfo uriInfo, @PathParam("folder") final String folderName,
+                                     @PathParam("path") final String pathParams, final Map<String, String> properties) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (pathParams, false, request, false, null);
 
-        if (visitor.isPresent() && visitor.get().getPersona() != null) {
-            persona = visitor.get().getPersona();
+        try {
+            final FileAsset vtlFile = getVTLFile(HTTPMethod.POST, request, folderName, initDataObject.getUser());
+            final Map<String, Object> contextParams = CollectionsUtils.map(
+                    "urlParams", initDataObject.getParamsMap(),
+                    "queryParams", uriInfo.getQueryParameters(),
+                    "postParams", properties,
+                    "user", initDataObject.getUser());
+
+            return evalVTLFile(request, response, vtlFile, contextParams,
+                    initDataObject.getUser(), getCacheStrategy(HTTPMethod.POST));
+
+        } catch(DotContentletStateException e) {
+            final String errorMessage = "Unable to find velocity file '" + HTTPMethod.POST.fileName + FILE_EXTENSION
+                    + "' under path '" + VTL_PATH + StringPool.SLASH + folderName + StringPool.SLASH + "'";
+            Logger.error(this, errorMessage, e);
+            return ResponseUtil.mapExceptionResponse(new DotDataException(errorMessage));
+        } catch(Exception e) {
+            Logger.error(this,"Exception on VTL endpoint. POST method: " + e.getMessage(), e);
+            return ResponseUtil.mapExceptionResponse(e);
         }
+    }
+
+    private Response evalVTLFile(final HttpServletRequest request, final HttpServletResponse response,
+                                 final FileAsset getFileAsset, final Map<String, Object> contextParams,
+                                 final User user, final DotJSONCacheStrategy cacheStrategy)
+            throws IOException {
+        final org.apache.velocity.context.Context context = VelocityUtil.getInstance().getContext(request, response);
+        contextParams.forEach(context::put);
+        context.put("dotJSON", new DotJSON());
+
+        final StringWriter evalResult = new StringWriter();
```

Comment:
It think it would be good to streaming the json esponse by using asynchronous responses such as the the LongPolling approach